### DISCUSSION TITLE
tRPC Benchmarks 📊

### DISCUSSION THREAD

**Discussion by jlalmes** [trpc/trpc#1962] [General]
## tRPC Benchmarks 📊

Would be good to grasp how tRPC performs against the raw underlying adapter frameworks. This info will be great for visibility so that we can focus our efforts on future performance improvements. 

The plan is to create test suites for each of the following frameworks with 1, 10 & 100 endpoints for each to identify how tRPC scales. Will initially cover `v9`, but will add `v10` test suites after release.

- [ ] `node:http`
- [ ] `Express`
- [ ] `Next.js`
- [ ] `Lambda`
- [ ] `Fastify`

Additional todo: Benchmarks for [`trpc-openapi`](https://github.com/jlalmes/trpc-openapi)

**Comment by KATT** [comment]
I played with this over a year ago: https://github.com/KATT/benchmarks

I don't see a reason why 1000 endpoints would be slower than 1 endpoint, they are stored in a flat `Record<string, Procedure>` in the top router.

**Comment by MarkLee425** [comment]
I am more curious for the comparison between fastify & trpc with fastify. Anyone has the latest update on this?
